plugin/src/main/java/org/opensearch/ml/task/MLTrainAndPredictTaskRunner.java

```diff
@@ -0,0 +1,172 @@
+        MLTask mlTask = MLTask
+            .builder()
+            .taskId(UUID.randomUUID().toString())
+            .taskType(MLTaskType.TRAINING_AND_PREDICTION)
+            .inputType(inputDataType)
+            .functionName(request.getMlInput().getFunctionName())
+            .state(MLTaskState.CREATED)
+            .workerNode(clusterService.localNode().getId())
+            .createTime(now)
+            .lastUpdateTime(now)
+            .async(false)
+            .build();
+        MLInput mlInput = request.getMlInput();
+
+        if (mlInput.getInputDataset().getInputDataType().equals(MLInputDataType.SEARCH_QUERY)) {
+            ActionListener<DataFrame> dataFrameActionListener = ActionListener
+                .wrap(dataFrame -> { trainAndPredict(mlTask, dataFrame, request, listener); }, e -> {
+                    log.error("Failed to generate DataFrame from search query", e);
+                    mlTaskManager.addIfAbsent(mlTask);
+                    handleMLTaskFailure(mlTask, e);
+                    listener.onFailure(e);
+                });
+            mlInputDatasetHandler
+                .parseSearchQueryInput(
+                    mlInput.getInputDataset(),
+                    new ThreadedActionListener<>(log, threadPool, TASK_THREAD_POOL, dataFrameActionListener, false)
+                );
+        } else {
+            DataFrame inputDataFrame = mlInputDatasetHandler.parseDataFrameInput(mlInput.getInputDataset());
+            threadPool.executor(TASK_THREAD_POOL).execute(() -> { trainAndPredict(mlTask, inputDataFrame, request, listener); });
+        }
+    }
+
+    private void trainAndPredict(
+        MLTask mlTask,
+        DataFrame inputDataFrame,
+        MLTrainingTaskRequest request,
+        ActionListener<MLPredictionTaskResponse> listener
+    ) {
+        // track ML task count and add ML task into cache
+        mlStats.getStat(ML_EXECUTING_TASK_COUNT.getName()).increment();
+        mlTaskManager.add(mlTask);
+
+        MLInput mlInput = request.getMlInput();
+
+        mlTaskManager.updateTaskState(mlTask.getTaskId(), MLTaskState.RUNNING, mlTask.isAsync());
+        // run predict
+        try {
+            mlTaskManager.updateTaskState(mlTask.getTaskId(), MLTaskState.RUNNING, mlTask.isAsync());
+            MLOutput output = MLEngine.trainAndPredict(mlInput.toBuilder().inputDataset(new DataFrameInputDataset(inputDataFrame)).build());
+            if (output instanceof MLPredictionOutput) {
+                ((MLPredictionOutput) output).setTaskId(mlTask.getTaskId());
+                ((MLPredictionOutput) output).setStatus(mlTaskManager.get(mlTask.getTaskId()).getState().name());
+            }
+
+            // Once prediction complete, reduce ML_EXECUTING_TASK_COUNT and update task state
+            handleMLTaskComplete(mlTask);
+            MLPredictionTaskResponse response = MLPredictionTaskResponse.builder().output(output).build();
+            listener.onResponse(response);
```

Comment:
Wondering if we should add a log info here showing the predict completion?

Reply:
sure, will add